libsolidity/ast/Types.cpp

```diff
@@ -198,7 +198,9 @@ TypePointer Type::forLiteral(Literal const& _literal)
 
 TypePointer Type::commonType(TypePointer const& _a, TypePointer const& _b)
 {
-	if (_b->isImplicitlyConvertibleTo(*_a))
+	if (!_a || !_b)
+		return TypePointer();
```

Comment:
Shall I understand `TypePointer()` as the type of every expression or the type of no expressions?  If there are no values that belong to `TypePointer()`, `TypePointer()` should be implicitly convertible to any type.  If all values belong to `TypePointer()`, every type should be implicitly convertible to `TypePointer()`.